public/javascripts/items.js

```diff
@@ -7,28 +7,105 @@ const fetchParams = {
   credentials: 'include'
 }
 
-const params = completed =>
-  Object.assign( {}, fetchParams, { body: JSON.stringify({ completed }) })
+const params = data =>
+  Object.assign( {}, FETCH_PARAMS, { body: JSON.stringify( data ) } )
 
-$(document).ready( () => {
-  $( '.title' ).click( event => {
-    const element = $( event.target )
-    const id = element.data( 'id' )
-    const completed = ! element.data( 'completed' )
+const RETURN_KEY = 13
+
+const checkJsonForSuccessField = json => {
+  if( json.success ) {
+    Promise.resolve( json )
+  } else {
+    Promise.reject( json.message )
+  }
+}
+
+const titleEdited = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+
+  if( event.charCode === RETURN_KEY ) {
+    let updatedTitle = element[0].value
+    fetch( `/items/${id}/edit`, params( { title: updatedTitle } ) )
+    .then( result => result.json() )
+    .then ( checkJsonForSuccessField )
+    .then( json => {
+      let span = $( `.title[data-id=${id}] > span`)
+      $( span[0] ).html(updatedTitle)
+      element.addClass( 'hidden')
+      $( span ).removeClass( 'hidden' )
+    })
+    .catch( error => {
+      console.log('error:', error)
+    })
+  }
+}
+
+const titleClicked = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+  $( element[0] ).addClass( 'hidden' )
+  let input = $( `.title[data-id=${id}] > input`)
+  $( input[0]).removeClass( 'hidden' )
+}
 
-    fetch( `/items/${id}/completed`, params( completed ))
+const descriptionEdited = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+
+  if( event.charCode === RETURN_KEY ) {
+    let updatedDescription = element[0].value
+    fetch( `/items/${id}/edit`, params( { description: updatedDescription } ) )
       .then( result => result.json() )
+      .then( checkJsonForSuccessField )
       .then( json => {
-        if( json.success ) {
-          element.data( 'completed', completed )
-
-          const parent = $( `.item[data-id=${id}]` )
-          if( completed ) {
-            parent.addClass( 'completed' )
-          } else {
-            parent.removeClass( 'completed' )
-          }
-        }
+        let span = $(`.description[data-id=${id}] > span`)
+        $( span[0] ).html( updatedDescription )
+        element.addClass( 'hidden')
+        $( span ).removeClass( 'hidden' )
```

Comment:
This block of code looks very similar to the block of code below (in `descriptionClicked`).  We can clean up our code and avoid duplication through the introduction of a function whose responsibility it is to hide and unhide things:
```js
const toggle = (className, id, text, parent) => {
  const element = $( `.${className}[data-id=${id}] > span` )
  
  // Do we need the [0] here?
  $( element[ 0 ] ).html( text )
  parent.addClass( 'hidden' )
  $( element ).removeClass( 'hidden' )
}
```
This section now becomes 
```js
.then( json => toggle( 'description', id, updatedDescription, element ) )
```
And successive blocks that do the same may be updated accordingly.